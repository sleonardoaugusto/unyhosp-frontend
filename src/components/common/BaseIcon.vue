<template>
  <div class="icon-wrapper">
    <svg class="icon" :width="width" :height="height">
      <use v-bind="{'xlink:href':'/feather-sprite.svg#'+name}"/>
    </svg>
  </div>
</template>

<script>
export default {
  props: {
    name: String,
    width: {
      type: [Number, String],
      default: 16
    },
    height: {
      type: [Number, String],
      default: 16
    }
  }
}
</script>

<style scoped>
  .icon-wrapper {
    display: inline-flex;
    align-items: center;
    color: #FFF;
    font-size: 1rem;
    font-weight: 600;
  }

  .icon {
    stroke: currentColor;
    stroke-width: 2;
    stroke-linecap: round;
    stroke-linejoin: round;
    fill: none;
    margin-right: 6px;
  }
</style>