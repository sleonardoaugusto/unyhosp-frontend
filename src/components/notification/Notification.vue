<template>
  <div class="notification-bar">
    <p class="notification-bar__message">
      <BaseIcon class="c-pointer" @click.native="close" name="x-circle"/>
      {{ notification.message }}
    </p>
  </div>
</template>

<script>
import BaseIcon from '@/components/common/BaseIcon'

export default {
  name: 'Notification',
  components: {
    BaseIcon
  },
  props: {
    notification: {
      type: Object,
      required: true
    }
  },
  methods: {
    close() {
      this.$emit('close-notification')
    }
  }
}
</script>

<style scoped>
  .c-pointer {
    cursor: pointer;
  }
</style>