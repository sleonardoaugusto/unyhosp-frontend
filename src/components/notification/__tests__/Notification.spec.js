import { shallowMount } from '@vue/test-utils'
import Notification from '@/components/notification/Notification'
import BaseIcon from '@/components/common/BaseIcon'

describe('Notification.vue', () => {
  let wrapper
  const close = jest.fn()

  beforeEach(() => {
    wrapper = shallowMount(Notification, {
      propsData: {
        notification: {
          message: 'Problema de conexÃ£o: Network Error',
          type: 'error'
        }
      }
    })
  })

  it('renders component', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('span click should trigger close method', () => {
    wrapper.setMethods({ close })
    wrapper.find(BaseIcon).trigger('click')
    expect(close).toHaveBeenCalled()
  })
})