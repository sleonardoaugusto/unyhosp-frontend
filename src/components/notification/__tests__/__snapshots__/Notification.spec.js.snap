// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notification.vue renders component 1`] = `
<div class="notification-bar error">
  <p class="notification-bar__message">
    <baseicon-stub name="x-circle" width="16" height="16" class="c-pointer"></baseicon-stub>
    Problema de conex√£o: Network Error
  </p>
</div>
`;
